codespan-reporting/src/term/views.rs

```diff
@@ -207,8 +211,21 @@ where
                 // 6 │ │     0 _ => "Fizz"
                 // 7 │ │     _ 0 => "Buzz"
                 // ```
-                // TODO(#125): If start line and end line are too far apart, add a source break.
                 for line_index in (start_line_index + 1)..end_line_index {
+                    if std::cmp::min(line_index - start_line_index, end_line_index - line_index)
+                        > renderer.context_lines()
+                        && !self.diagnostic.labels.iter().any(|label| {
+                            // check that no other label is on this line
+                            start_line_range.start <= label.range.start
+                                && label.range.start <= end_line_range.end
+                                || start_line_range.start <= label.range.end
+                                    && label.range.end <= end_line_range.end
+                        })
```

Comment:
Perhaps instead we could do `label.single_lines.is_empty()`? This avoids iterating over all the lines in the diagnostic again.